services/catalog/tests/test_endpoints.py

```diff
@@ -333,6 +330,152 @@ def test_search_with_collections_and_filter(self, client):  # pylint: disable=mi
         response = client.post("/catalog/search", json=test_json)
         assert response.status_code == 200
 
+    @pytest.mark.parametrize(
+        "method",
+        ["POST", "GET"],
+    )
+    def test_search_using_several_collections_post(self, client, method):
+        """Test a search request involving several collections (POST method)"""
+        # Search items on several collections without using implicit naming feature
+        test_json = {
+            "collections": ["toto_S1_L1", "toto_S2_L3"],
+            "filter": {
+                "op": "and",
+                "args": [
+                    {"op": "=", "args": [{"property": "width"}, 2500]},
+                    {"op": "=", "args": [{"property": "height"}, 2500]},
+                ],
+            },
+        }
+        response = client.post("/catalog/search", json=test_json)
+
+        assert response.status_code == fastapi.status.HTTP_200_OK
+        assert len(json.loads(response.content)["features"]) == 3
+
+        # Use implicit naming mechanism for some collections of the list + specify owner in the content/query parameters
+        if method == "POST":
+            test_json = {
+                "collections": ["S1_L1", "S2_L3"],
+                "owner": "toto",
+                "filter": {
+                    "op": "and",
+                    "args": [
+                        {"op": "=", "args": [{"property": "width"}, 2500]},
+                        {"op": "=", "args": [{"property": "height"}, 2500]},
+                    ],
+                },
+            }
+            response = client.post("/catalog/search", json=test_json)
+        elif method == "GET":
+            test_params = {
+                "collections": "S1_L1,toto_S2_L3",
+                "filter-lang": "cql2-text",
+                "filter": "width=2500",
+                "owner": "toto",
+            }
+            response = client.get("/catalog/search", params=test_params)
+        assert response.status_code == fastapi.status.HTTP_200_OK
+        assert len(json.loads(response.content)["features"]) == 3
+
+        # Use implicit naming mechanism for some collections of the list + specify owner in the filter
```

Comment:
You don't specify owner in json as described in comment above

Reply:
Here he owner_id is specified in the body (POST method) or in the query parameters (GET method)